src/Shared/TransportationDeviceClient.cs

```diff
@@ -79,9 +76,36 @@ public async Task<dynamic> GetDynamicDigitalTwinAsync()
             return dyn;
         }
 
-        public Task RegisterDesiredPropertyUpdateCallbackAsync(MethodCallback methodHandler)
+        // this method will process any desired property change events
+        private static async Task OnDesiredPropertyChanged(TwinCollection desiredProperties, object userContext)
```

Comment:
no await in the async method, is that okay ?

Reply:
There's a "waitall" on the task list execution.

Reply:
wait/waitall are different than await in how they work underneath.

~~~~
        private static async Task DoSomethingAsync()
        {
            Task.Run(()=> Console.WriteLine("hello from async method")).Wait();
        }
~~~~
that code still gives a warning: `This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.`